src/sentry/runner/commands/upgrade.py

```diff
@@ -1,12 +1,45 @@
 import click
 from django.conf import settings
+from django.db import connections
 
 from sentry.runner.decorators import configuration
 
 
+def _check_history():
+    connection = connections["default"]
+    cursor = connection.cursor()
+    try:
+        # If the database has 0 tables we are good to go.
+        cursor.execute(
+            """
+            SELECT COUNT(*) AS total FROM information_schema.tables
+            WHERE table_catalog = %s AND table_schema = 'public'
+            """,
+            [settings.DATABASES["default"]["NAME"]],
+        )
+        if cursor.fetchone()[0] == 0:
+            return
+    except Exception:
+        raise click.ClickException("Could not determine migration state. Aborting")
+
+    try:
+        # If we haven't run all the migration up to the latest squash abort.
+        # As we squash more history this should be updated.
+        cursor.execute("SELECT 1 FROM django_migrations WHERE name = '0200_release_indices'")
+        if not cursor.fetchone()[0]:
+            raise RuntimeError
+    except Exception:
+        raise click.ClickException(
```

Comment:
Similar note as with the previous try-except. I feel like this can swallow other exceptions and then the message about the upgrade process will be misleading. I'd suggest here to raise the `ClickException` right away instead of the `RuntimeError` and remove the broad try-except block to let any other exception bubble up.